File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/teacher/TeacherService.java
package kh.farrukh.sortmapperexamplejava.teacher;

import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherCreateRequestDTO;
import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherDetailsResponseDTO;
import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherResponseDTO;
import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherUpdateRequestDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TeacherService {

  void createTeacher(TeacherCreateRequestDTO requestDTO);

  void updateTeacher(Long id, TeacherUpdateRequestDTO requestDTO);

  void deleteTeacher(Long id);

  TeacherDetailsResponseDTO getTeacher(Long id);

  Page<TeacherResponseDTO> getTeachers(Pageable pageable);

  TeacherEntity findTeacher(Long id);

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/teacher/TeacherService.kt
package kh.farrukh.sortmapperexample.teacher

import kh.farrukh.sortmapperexample.teacher.model.TeacherCreateRequestDTO
import kh.farrukh.sortmapperexample.teacher.model.TeacherDetailsResponseDTO
import kh.farrukh.sortmapperexample.teacher.model.TeacherResponseDTO
import kh.farrukh.sortmapperexample.teacher.model.TeacherUpdateRequestDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface TeacherService {

    fun createTeacher(requestDTO: TeacherCreateRequestDTO)

    fun getTeacherById(id: Long): TeacherDetailsResponseDTO

    fun getTeachers(pageable: Pageable): Page<TeacherResponseDTO>

    fun updateTeacher(id: Long, requestDTO: TeacherUpdateRequestDTO)

    fun deleteTeacher(id: Long)

}